libbeat/publisher/queue/diskqueue/consumer.go

```diff
@@ -0,0 +1,52 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package diskqueue
+
+import (
+	"github.com/elastic/beats/v7/libbeat/publisher"
+	"github.com/elastic/beats/v7/libbeat/publisher/queue"
+)
+
+type diskQueueConsumer struct {
+}
+
+type diskQueueBatch struct {
+}
+
+//
+// diskQueueConsumer implementation of the queue.Consumer interface
+//
+
+func (consumer *diskQueueConsumer) Get(eventCount int) (queue.Batch, error) {
+	return &diskQueueBatch{}, nil
```

Comment:
If it's not implemented consider to use `panic("TODO: not implemented")`. No need to return values if `panic` is used. Plus you can grep the code for `TODO` in case you might have missed something.